src/components/Calendar/doubleCalendar/index.js

```diff
@@ -0,0 +1,350 @@
+import React, { useMemo, useState, useCallback, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { useUniqueIdentifier } from '../../../libs/hooks';
+import {
+    UP_KEY,
+    DOWN_KEY,
+    RIGHT_KEY,
+    LEFT_KEY,
+    HOME_KEY,
+    END_KEY,
+    PAGEUP_KEY,
+    PAGEDN_KEY,
+    SPACE_KEY,
+    ENTER_KEY,
+} from '../../../libs/constants';
+import { Provider } from '../context';
+import RightIcon from '../icons/rightArrow';
+import LeftIcon from '../icons/leftArrow';
+import DaysOfWeek from '../daysOfWeek';
+import Month from '../month';
+import {
+    normalizeDate,
+    getFirstDayMonth,
+    addMonths,
+    getNextFocusedDate,
+    isSameMonth,
+    isEmptyRange,
+    isDateBelowLimit,
+} from '../helpers';
+import {
+    useYearsRange,
+    useLastDayMonth,
+    useMoveFocusedDayFunction,
+    useMoveFocusedMonthFunction,
+} from './hooks';
+import StyledControlsContainer from '../styled/controlsContainer';
+import StyledArrowButton from '../styled/arrowButton';
+import StyledMonthsContainer from './styled/monthsContainer';
+import StyledTable from '../styled/table';
+import StyledContainer from './styled/container';
+import StyledCalendar from './styled/calendar';
+import StyledDivider from './styled/divider';
+import MonthHeader from './monthHeader';
+
+export default function DoubleCalendar(props) {
+    const {
+        id,
+        className,
+        style,
+        locale,
+        value,
+        minDate,
+        maxDate,
+        onChange,
+        selectedRange,
+        selectionType,
+    } = props;
+    const [focusedDate, setFocusedDate] = useState(normalizeDate(value));
+    const [currentMonth, setCurrentMonth] = useState(getFirstDayMonth(normalizeDate(value)));
+    const [currentRange, setCurrentRange] = useState(selectedRange);
+    const [enableNavKeys, setEnableNavKeys] = useState(false);
+
+    const rightCalendarMonth = addMonths(currentMonth, 1);
+    const firstMonthLabelId = useUniqueIdentifier('first-month');
+    const secondMonthLabelId = useUniqueIdentifier('second-month');
+    const currentYear = currentMonth.getFullYear();
+    const rightCalendarYear = rightCalendarMonth.getFullYear();
+    const yearsRange = useYearsRange(minDate, maxDate, currentMonth);
+    const lastYearItem = yearsRange[yearsRange.length - 1];
+    const minSelectableDate = minDate || new Date(yearsRange[0].value, 0, 1);
+    const maxSelectableDate = maxDate || new Date(lastYearItem.value, 11, 31);
+    const lastDayMonth = useLastDayMonth(currentMonth);
+    const disableNextMonth = addMonths(rightCalendarMonth, 1) > maxSelectableDate;
+    const disablePreviousMonth = lastDayMonth < minSelectableDate;
```

Comment:
I think you should put all this in a custom Hook useDisabledControls that returns [isDisableNextMonth, isDisablePreviousMonth] and useLastDayMonth is not necessary because it will only be used inside useDisabledControls